substrate/frame/revive/src/wasm/runtime.rs

```diff
@@ -1676,6 +1679,25 @@ pub mod env {
 		)?)
 	}
 
+	/// Stores the current block author into the supplied buffer.
+	/// See [`pallet_revive_uapi::HostFn::block_author`].
+	#[stable]
+	fn block_author(&mut self, memory: &mut M, out_ptr: u32) -> Result<(), TrapReason> {
+		self.charge_gas(RuntimeCosts::BlockAuthor)?;
+		let block_author = self
+			.ext
+			.block_author()
+			.map(|account| <E::T as Config>::AddressMapper::to_address(&account))
+			.unwrap_or(H160::zero());
```

Comment:
Does this ever happen on Eth?

Reply:
I don't see anything against it: Whoever implements this can return a hard-coded address anyways if they wish to do so.

Reply:
Okay then we leave it as-is.